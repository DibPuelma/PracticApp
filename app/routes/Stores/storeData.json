{
  "stores": [
    "http://www.borde.cl/wp-content/uploads/2005/11/Americanino-2.png",
    "http://zapalook.com.ar/wp-content/uploads/2015/07/logo-Amphora.jpg",
    "http://static03.ofertia.cl/comercios/Umbrale/profile-850.v11.png",
    "http://www.brycomm.com/wp-content/uploads/2011/08/Samsung-Logo.png",
    "http://pelfind.me/images/187941/apple-logo-9th.jpg",
    "http://emblemsbf.com/img/7808.jpg",
    "https://www.republicamoda.com/media/catalog/category/20120302-logoPrune_1.jpg",
    "http://static.dnaindia.com/sites/default/files/2015/04/19/328875-reebok-logo.jpg",
    "http://www.tutarjeta.cl/wp-content/uploads/2016/01/logo-microplay.jpg"
  ]
}
