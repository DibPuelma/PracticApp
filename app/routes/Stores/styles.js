import {
  StyleSheet,
  Dimensions
} from 'react-native';

module.exports = StyleSheet.create({
  loadingContainer: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center', 
    justifyContent:'center',
    marginTop: 72,
    marginLeft: 10,
    marginBottom: 10,
    marginRight: 10
  },
  scrollview: {
    flex: 1,
    marginLeft: 10,
    marginTop: 72,
    marginRight: 10,
    marginBottom: 10
  },
  container: {
    elevation: 3
  },
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
    flexWrap: 'wrap',
  },
  logo: {
    width: 100,
    height: 100,
    margin: 10
  }
});
