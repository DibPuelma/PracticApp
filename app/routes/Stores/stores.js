import React, { Component } from 'react';
import {
  View,
  Image,
  Text,
  ScrollView,
  ActivityIndicator
} from 'react-native';

import storeData from './storeData';
import styles from './styles';
import backButtonHandler from '../../lib/backButtonHandler';
import settings from '../../config/settings';

var StoresStatus = {
  WAITING: 'waiting',
  READY  : 'ready'
};

export default class Stores extends Component {

  constructor(props) {
    super(props);
    this._backToPrevious = this._backToPrevious.bind(this);
    this.singletonBackButtonHandler = backButtonHandler.getInstance();
    this.state = { 
      status: StoresStatus.WAITING,
      storeData: []
    };
  }

  componentDidMount() {
    this._companiesRequest();
  }

  componentWillMount() {
    this.singletonBackButtonHandler.addBackEvent(this._backToPrevious);
  }

  componentWillUnmount() {
    this.singletonBackButtonHandler.removeBackEvent(this._backToPrevious);
  }

  _backToPrevious() {
    this.props.navigator.replace({id:'QRReader'});
    return true; // This is important to prevent multiple calls
  }

  render() {
    if (this.state.status === StoresStatus.WAITING) {
      return (
      <View style={styles.loadingContainer}>
        <ActivityIndicator animating={true} style={[styles.centering, {height: 50}]} size="large" color="#3FA9F5"/>
      </View>
      )
    }
    return(
      <ScrollView style={styles.scrollview}>
        <View style={styles.container}>
        { this.state.storeData.length > 0 &&
          this.state.storeData.map((store, i) => (
            <View key={i} style={styles.logo}>
              <Image source={{ uri: store.logo }} style={styles.logo} />
            </View>
          ))
        }
        </View>
      </ScrollView>
    );
  }

  _companiesRequest() {
    var url = settings.COMPANIES_REQUEST;
    var promise = fetch(url, {
      method: 'GET',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
      }, 
    })
    .then((response) => response.json());
    
    var stores = this;

    promise.then(function(result) {
      stores.setState({ status: StoresStatus.READY , storeData: result });
    }, function(err) { // error
      console.log(err);    
    });
  }
}
