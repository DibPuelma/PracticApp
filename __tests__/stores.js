import 'react-native';
import React from 'react';
import Stores from '../app/routes/Stores/stores'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Stores />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
