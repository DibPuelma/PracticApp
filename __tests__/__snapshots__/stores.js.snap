exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#E6E6E6",
      "borderColor": "#C6C6C6",
      "borderWidth": 1,
      "elevation": 3,
      "flex": 1,
      "height": 600,
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 72,
    }
  }>
  <ScrollView
    contentContainerStyle={
      Object {
        "backgroundColor": "#E6E6E6",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }>
    <View
      style={
        Object {
          "height": 120,
          "margin": 10,
          "width": 120,
        }
      }>
      <Image
        source={
          Object {
            "uri": "http://www.borde.cl/wp-content/uploads/2005/11/Americanino-2.png",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 10,
            "width": 120,
          }
        } />
    </View>
    <View
      style={
        Object {
          "height": 120,
          "margin": 10,
          "width": 120,
        }
      }>
      <Image
        source={
          Object {
            "uri": "http://zapalook.com.ar/wp-content/uploads/2015/07/logo-Amphora.jpg",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 10,
            "width": 120,
          }
        } />
    </View>
    <View
      style={
        Object {
          "height": 120,
          "margin": 10,
          "width": 120,
        }
      }>
      <Image
        source={
          Object {
            "uri": "http://static03.ofertia.cl/comercios/Umbrale/profile-850.v11.png",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 10,
            "width": 120,
          }
        } />
    </View>
    <View
      style={
        Object {
          "height": 120,
          "margin": 10,
          "width": 120,
        }
      }>
      <Image
        source={
          Object {
            "uri": "http://www.brycomm.com/wp-content/uploads/2011/08/Samsung-Logo.png",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 10,
            "width": 120,
          }
        } />
    </View>
    <View
      style={
        Object {
          "height": 120,
          "margin": 10,
          "width": 120,
        }
      }>
      <Image
        source={
          Object {
            "uri": "http://pelfind.me/images/187941/apple-logo-9th.jpg",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 10,
            "width": 120,
          }
        } />
    </View>
    <View
      style={
        Object {
          "height": 120,
          "margin": 10,
          "width": 120,
        }
      }>
      <Image
        source={
          Object {
            "uri": "http://emblemsbf.com/img/7808.jpg",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 10,
            "width": 120,
          }
        } />
    </View>
    <View
      style={
        Object {
          "height": 120,
          "margin": 10,
          "width": 120,
        }
      }>
      <Image
        source={
          Object {
            "uri": "https://www.republicamoda.com/media/catalog/category/20120302-logoPrune_1.jpg",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 10,
            "width": 120,
          }
        } />
    </View>
    <View
      style={
        Object {
          "height": 120,
          "margin": 10,
          "width": 120,
        }
      }>
      <Image
        source={
          Object {
            "uri": "http://static.dnaindia.com/sites/default/files/2015/04/19/328875-reebok-logo.jpg",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 10,
            "width": 120,
          }
        } />
    </View>
    <View
      style={
        Object {
          "height": 120,
          "margin": 10,
          "width": 120,
        }
      }>
      <Image
        source={
          Object {
            "uri": "http://www.tutarjeta.cl/wp-content/uploads/2016/01/logo-microplay.jpg",
          }
        }
        style={
          Object {
            "height": 120,
            "margin": 10,
            "width": 120,
          }
        } />
    </View>
  </ScrollView>
</View>
`;
